conceptual/CustomPatterns/Aspects/Applying/extension-blocks-multicast.md

```diff
@@ -0,0 +1,179 @@
+
+
+## What Are Extension Blocks?
+
+C# 14 introduces a new syntax for declaring extension members using `extension` blocks within static classes. Extension blocks allow you to define extension methods and properties, including static extension members that appear to extend the type itself rather than instances.
+
+Example:
+
+```csharp
+public static class TestClassExtensions
+{
+    extension<TInstance>(TInstance instance)
+    {
+        // Instance extension property
+        public TInstance ExtensionProperty
+        {
+            get => instance;
+            set { }
+        }
+
+        // Static extension property
+        public static int StaticExtensionProperty
+        {
+            get => 42;
+            set { }
+        }
+
+        // Instance extension method
+        public TInstance ExtensionMethod() => instance;
+
+        // Static extension method
+        public static void StaticExtensionMethod() { }
+    }
+}
+```
+
+
+## How Extension Blocks Are Implemented in IL
+
+In C# 14, extension blocks are implemented by the compiler as static methods and a set of special metadata types. These metadata types are intended for the C# compiler to match extension methods and properties with the receiver type. Since both the implementation methods and the metadata types may not be expected by existing aspects, PostSharp 2026.0 introduces explicit opt-in support into the multicast engine.
+
+
+## Applying Aspects to Extension Block Members
+
+By default, PostSharp does not apply aspects or multicast attributes to members declared within extension blocks. This is a safety measure to avoid unintended behavior, as extension block members have unique characteristics that differ from regular methods and properties.
+
+Extension block metadata are always skipped by multicasting, but may still be targeted by <xref:PostSharp.Aspects.Advices.IAdviceProvider> and <xref:PostSharp.Aspects.IAspectProvider> by mistake. Doing this will result in an error.
+
+### Enabling Extension Block Support
+
+To apply aspects to extension block members, set the `AllowExtensionBlockMembers` property to `true`. This property is available on:
+
+- <xref:PostSharp.Extensibility.MulticastAttribute.AllowExtensionBlockMembers> - Set this property when applying the aspect to control whether it targets extension block members.
+- <xref:PostSharp.Extensibility.MulticastAttributeUsageAttribute.AllowExtensionBlockMembers> - Set this property on the aspect class definition to change the default behavior.
+
+
+### Example: Applying Aspects to Extension Methods
+
+```csharp
+[PSerializable]
+class LoggingAspect : OnMethodBoundaryAspect
+{
+    public override void OnEntry(MethodExecutionArgs args)
+    {
+        Console.WriteLine($"Entering: {args.Method.Name}");
+    }
+}
+
+public static class TestClassExtensions
+{
+    extension<TInstance>(TInstance instance)
+    {
+        [LoggingAspect(AllowExtensionBlockMembers = true)]
```

Comment:
This is rather a bug. When the attribute is directly applied to an extension block member, AllowExtensionBlockMembers  should not be required. It makes sense only when multicasting from the type